Linux arm64 deb points to x86_64 pty.node

Fresh install is not working with version 6.5.2-309-309

```
> open-lens   
innerError Error: Cannot find module '../build/Debug/pty.node'
Require stack:
- /opt/OpenLens/resources/app.asar/node_modules/node-pty/lib/unixTerminal.js
- /opt/OpenLens/resources/app.asar/node_modules/node-pty/lib/index.js
- /opt/OpenLens/resources/app.asar/node_modules/@k8slens/core/static/build/library/main.js
- /opt/OpenLens/resources/app.asar/static/build/main.js
- 
    at Module._resolveFilename (node:internal/modules/cjs/loader:963:15)
    at n._resolveFilename (node:electron/js2c/browser_init:2:109751)
    at Module._load (node:internal/modules/cjs/loader:811:27)
    at f._load (node:electron/js2c/asar_bundle:2:13330)
    at Module.require (node:internal/modules/cjs/loader:1035:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/opt/OpenLens/resources/app.asar/node_modules/node-pty/lib/unixTerminal.js:30:15)
    at Module._compile (node:internal/modules/cjs/loader:1141:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1196:10)
    at Module.load (node:internal/modules/cjs/loader:1011:32) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/opt/OpenLens/resources/app.asar/node_modules/node-pty/lib/unixTerminal.js',
    '/opt/OpenLens/resources/app.asar/node_modules/node-pty/lib/index.js',
    '/opt/OpenLens/resources/app.asar/node_modules/@k8slens/core/static/build/library/main.js',
    '/opt/OpenLens/resources/app.asar/static/build/main.js',
    undefined
  ]
}
A JavaScript error occurred in the main process
Uncaught Exception:
Error: /opt/OpenLens/resources/app.asar.unpacked/node_modules/node-pty/build/Release/pty.node: cannot open shared object file: No such file or directory
    at process.func [as dlopen] (node:electron/js2c/asar_bundle:2:1822)
    at Module._extensions..node (node:internal/modules/cjs/loader:1226:18)
    at Object.func [as .node] (node:electron/js2c/asar_bundle:2:2049)
    at Module.load (node:internal/modules/cjs/loader:1011:32)
    at Module._load (node:internal/modules/cjs/loader:846:12)
    at f._load (node:electron/js2c/asar_bundle:2:13330)
    at Module.require (node:internal/modules/cjs/loader:1035:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/opt/OpenLens/resources/app.asar/node_modules/node-pty/lib/unixTerminal.js:26:11)
    at Module._compile (node:internal/modules/cjs/loader:1141:14)
[193532:0607/233710.223407:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: No such file or directory (2)
[193532:0607/233710.223555:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: No such file or directory (2)
[193532:0607/233710.223647:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: No such file or directory (2)
[193532:0607/233710.223677:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: No such file or directory (2)
[193532:0607/233710.223697:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: No such file or directory (2)
[193532:0607/233710.223717:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: No such file or directory (2)
[193532:0607/233710.223735:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: No such file or directory (2)
[193532:0607/233710.224444:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: No such file or directory (2)
[193532:0607/233710.224464:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: No such file or directory (2)
[193532:0607/233710.224483:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: No such file or directory (2)
[193532:0607/233710.224503:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: No such file or directory (2)
[193532:0607/233710.224523:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: No such file or directory (2)
[193532:0607/233710.224541:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: No such file or directory (2)
[193532:0607/233710.224560:ERROR:gbm_wrapper.cc(255)] Failed to export buffer to dma_buf: No such file or directory (2)
```

File is present but it's built against x86_64 instead of target arch, Linux arm64 in this case:
```─> ldd /opt/OpenLens/resources/app.asar.unpacked/node_modules/node-pty/build/Release/pty.node                                                                        
        not a dynamic executable

─> file /opt/OpenLens/resources/app.asar.unpacked/node_modules/node-pty/build/Release/pty.node    
/opt/OpenLens/resources/app.asar.unpacked/node_modules/node-pty/build/Release/pty.node: ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, BuildID[sha1]=eeb4c2c5dd10b9aaf3d07790fe81f90565ba2a34, not stripped
```

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -56,6 +56,8 @@ jobs:
           if [ "$RUNNER_OS" == "Windows" ]; then
             npx nx run open-lens:build:app --win nsis msi portable
           elif [ "$RUNNER_OS" == "Linux" ]; then
+            aarch64-linux-gnu-g++ --version
+            exit 1
             cd open-lens
             env npm_config_target_arch=arm64 npm_config_arch=arm64 npm rebuild --loglevel silly node-pty
             find . -name '*pty.node' -print0 | xargs -0 file

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -56,11 +56,7 @@ jobs:
           if [ "$RUNNER_OS" == "Windows" ]; then
             npx nx run open-lens:build:app --win nsis msi portable
           elif [ "$RUNNER_OS" == "Linux" ]; then
-            npx nx run open-lens:build:app --x64
-            rm -rf node_modules
-            npm run all:install -- --arch=arm64
-            npx nx reset
-            npx nx run open-lens:build:app --arm64
+            npx nx run open-lens:build:app --x64 --arm64
             find . -name '*pty.node' -print0 | xargs -0 file
           else
             npx nx run open-lens:build:app --x64 --arm64
@@ -82,7 +78,6 @@ jobs:
           WIN_CSC_KEY_PASSWORD: ${{ secrets.CSC_KEY_PASSWORD_WIN }}
           CSC_FOR_PULL_REQUEST: true
           DOWNLOAD_ALL_ARCHITECTURES: ${{ runner.os != 'Windows' }}
-          DEBUG: electron-rebuild
 
       - name: Calculate SHA256 checksum
         run: |-

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -11,6 +11,7 @@ on:
 jobs:
   build:
     strategy:
+      fail-fast: false
       matrix:
         os: [ubuntu-20.04, macos-latest, windows-latest]
     runs-on: ${{ matrix.os }}

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -56,14 +56,14 @@ jobs:
           if [ "$RUNNER_OS" == "Windows" ]; then
             npx nx run open-lens:build:app --win nsis msi portable
           elif [ "$RUNNER_OS" == "Linux" ]; then
-            npx nx run open-lens:build:app --x64 --arm64
+            npx nx run open-lens:build:app --x64
             rm -rf node_modules
             npm run all:install -- --arch=arm64
             find . -name '*pty.node' -print0 | xargs -0 file
             env npm_config_target_arch=arm64 npm_config_arch=arm64 npm rebuild --verbose node-pty@0.10.1
             find . -name '*pty.node' -print0 | xargs -0 file
             npx nx reset
-            npx nx run open-lens:build:app 
+            npx nx run open-lens:build:app --arm64
             find . -name '*pty.node' -print0 | xargs -0 file
           else
             npx nx run open-lens:build:app --x64 --arm64

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -59,6 +59,8 @@ jobs:
             env npm_config_target_arch=arm64 npm_config_arch=arm64 npm rebuild --loglevel debug node-pty@0.10.1
             env npm_config_target_arch=arm64 npm_config_arch=arm64 npm rebuild -w open-lens --loglevel debug node-pty@0.10.1
             find . -name '*pty.node' -print0 | xargs -0 file
+            env npm_config_target_arch=arm64 npm_config_arch=arm64 npx @electron/rebuild -f
+            find . -name '*pty.node' -print0 | xargs -0 file
             env npm_config_target_arch=arm64 npm_config_arch=arm64 npx @electron/rebuild -f -a arm64
             find . -name '*pty.node' -print0 | xargs -0 file
             cd open-lens

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -13,7 +13,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [ubuntu-20.04]
+        os: [ubuntu-20.04, macos-latest, windows-latest]
     runs-on: ${{ matrix.os }}
     timeout-minutes: 360
     env:

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -56,13 +56,8 @@ jobs:
           if [ "$RUNNER_OS" == "Windows" ]; then
             npx nx run open-lens:build:app --win nsis msi portable
           elif [ "$RUNNER_OS" == "Linux" ]; then
-            env npm_config_target_arch=arm64 npm_config_arch=arm64 npm rebuild --verbose node-pty@0.10.1
-            find . -name '*pty.node' -print0 | xargs -0 file
             env npm_config_target_arch=arm64 npm_config_arch=arm64 npx --yes --package npm@^9.6.7 npm rebuild --debug node-pty@0.10.1
-            find . -name '*pty.node' -print0 | xargs -0 file
-            env npm_config_target_arch=aarch64 npm_config_arch=aarch64 npm rebuild --verbose node-pty@0.10.1
-            find . -name '*pty.node' -print0 | xargs -0 file
-            env npm_config_target_arch=aarch64 npm_config_arch=aarch64 npx --yes --package npm@^9.6.7 npm rebuild --debug node-pty@0.10.1
+            env npm_config_target_arch=arm64 npm_config_arch=arm64 npx --yes --package npm@^9.6.7 npm rebuild -w open-lens --debug node-pty@0.10.1
             find . -name '*pty.node' -print0 | xargs -0 file
             env npm_config_target_arch=arm64 npm_config_arch=arm64 npx @electron/rebuild -f -a amd64
             find . -name '*pty.node' -print0 | xargs -0 file

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -61,7 +61,11 @@ jobs:
             find . -name '*pty.node' -print0 | xargs -0 file
             env npm_config_target_arch=arm64 npm_config_arch=arm64 npx @electron/rebuild -f -a amd64
             find . -name '*pty.node' -print0 | xargs -0 file
-            exit 0
+            cd open-lens
+            env npm_config_target_arch=arm64 npm_config_arch=arm64 npx @electron/rebuild -f -a amd64
+            cd ..
+            find . -name '*pty.node' -print0 | xargs -0 file
+            exit 1
             
           
             npx nx run open-lens:build:app --x64

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -59,10 +59,10 @@ jobs:
             env npm_config_target_arch=arm64 npm_config_arch=arm64 npm rebuild --loglevel debug node-pty@0.10.1
             env npm_config_target_arch=arm64 npm_config_arch=arm64 npm rebuild -w open-lens --loglevel debug node-pty@0.10.1
             find . -name '*pty.node' -print0 | xargs -0 file
-            env npm_config_target_arch=arm64 npm_config_arch=arm64 npx @electron/rebuild -f -a amd64
+            env npm_config_target_arch=arm64 npm_config_arch=arm64 npx @electron/rebuild -f -a arm64
             find . -name '*pty.node' -print0 | xargs -0 file
             cd open-lens
-            env npm_config_target_arch=arm64 npm_config_arch=arm64 npx @electron/rebuild -f -a amd64
+            env npm_config_target_arch=arm64 npm_config_arch=arm64 npx @electron/rebuild -f -a arm64
             cd ..
             find . -name '*pty.node' -print0 | xargs -0 file
             exit 1

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -60,7 +60,7 @@ jobs:
             find . -name '*pty.node' -print0 | xargs -0 file
             env npm_config_target_arch=arm64 npm_config_arch=arm64 npx --yes --package npm@^9.6.7 npm rebuild --verbose node-pty@0.10.1
             find . -name '*pty.node' -print0 | xargs -0 file
-            env npm_config_target_arch=arm64 npm_config_arch=arm64 npx @electro/rebuild -f -a amd64
+            env npm_config_target_arch=arm64 npm_config_arch=arm64 npx @electron/rebuild -f -a amd64
             find . -name '*pty.node' -print0 | xargs -0 file
             exit 0
             

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -82,6 +82,7 @@ jobs:
           WIN_CSC_KEY_PASSWORD: ${{ secrets.CSC_KEY_PASSWORD_WIN }}
           CSC_FOR_PULL_REQUEST: true
           DOWNLOAD_ALL_ARCHITECTURES: ${{ runner.os != 'Windows' }}
+          DEBUG: electron-rebuild
 
       - name: Calculate SHA256 checksum
         run: |-

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -42,7 +42,7 @@ jobs:
 
       - name: Build Lens
         run: |
-          
+          npm run all:install
           # Avoid npm install from electron-builder
           # ref: https://github.com/electron-userland/electron-builder/blob/80a3ae857b9305aff1774e1b2b932601af50231b/packages/app-builder-lib/src/util/yarn.ts#L11-L16
           mkdir -p open-lens/node_modules
@@ -57,7 +57,7 @@ jobs:
           elif [ "$RUNNER_OS" == "Linux" ]; then
             sudo apt update
             sudo apt install gcc-aarch64-linux-gnu g++-aarch64-linux-gnu -y          
-            env CC=aarch64-linux-gnu-gcc CXX=aarch64-linux-gnu-g++ npm_config_target_arch=arm64 npm_config_arch=arm64 npm run all:install
+            export CC=aarch64-linux-gnu-gcc CXX=aarch64-linux-gnu-g++
             npx nx run open-lens:build:app --arm64
             find . -name '*pty.node' -print0 | xargs -0 file
             exit 1

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -42,7 +42,7 @@ jobs:
 
       - name: Build Lens
         run: |
-          npm run all:install
+          npm run all:install -- --arch=arm64
           # Avoid npm install from electron-builder
           # ref: https://github.com/electron-userland/electron-builder/blob/80a3ae857b9305aff1774e1b2b932601af50231b/packages/app-builder-lib/src/util/yarn.ts#L11-L16
           mkdir -p open-lens/node_modules

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -12,7 +12,7 @@ jobs:
   build:
     strategy:
       matrix:
-        os: [ubuntu-20.04, macos-latest, windows-latest]
+        os: [ubuntu-20.04]
     runs-on: ${{ matrix.os }}
     timeout-minutes: 360
     env:
@@ -55,13 +55,16 @@ jobs:
           if [ "$RUNNER_OS" == "Windows" ]; then
             npx nx run open-lens:build:app --win nsis msi portable
           elif [ "$RUNNER_OS" == "Linux" ]; then
-            npx nx run open-lens:build:app --x64
-          
             sudo apt update
-            sudo apt install gcc-aarch64-linux-gnu g++-aarch64-linux-gnu -y  
-            export CC=aarch64-linux-gnu-gcc CXX=aarch64-linux-gnu-g++ npm_config_target_arch=arm64 npm_config_arch=arm64
+            sudo apt install gcc-aarch64-linux-gnu g++-aarch64-linux-gnu -y          
+            export CC=aarch64-linux-gnu-gcc CXX=aarch64-linux-gnu-g++ npm_config_target_arch=arm64 npm_config_arch=arm64 
+            npx nx run open-lens:build:app --arm64
+            find . -name '*pty.node' -print0 | xargs -0 file
+            exit 1
           
-            npm run all:reinstall
+            
+          
+            env npm_config_target_arch=arm64 npm_config_arch=arm64 npm run all:reinstall
             npx nx reset
             npx nx run open-lens:build:app --arm64
             find . -name '*pty.node' -print0 | xargs -0 file

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -12,7 +12,7 @@ jobs:
   build:
     strategy:
       matrix:
-        os: [ubuntu-20.04]
+        os: [ ubuntu-20.04, macos-latest, windows-latest ]
     runs-on: ${{ matrix.os }}
     timeout-minutes: 360
     env:
@@ -55,17 +55,14 @@ jobs:
           if [ "$RUNNER_OS" == "Windows" ]; then
             npx nx run open-lens:build:app --win nsis msi portable
           elif [ "$RUNNER_OS" == "Linux" ]; then
+            npx nx run open-lens:build:app          
+            npm run all:reinstall
+            npx nx reset
+          
             sudo apt update
             sudo apt install gcc-aarch64-linux-gnu g++-aarch64-linux-gnu -y          
             export CC=aarch64-linux-gnu-gcc CXX=aarch64-linux-gnu-g++
-            npx nx run open-lens:build:app --arm64
-            find . -name '*pty.node' -print0 | xargs -0 file
-            exit 1
           
-            
-          
-            env npm_config_target_arch=arm64 npm_config_arch=arm64 npm run all:reinstall
-            npx nx reset
             npx nx run open-lens:build:app --arm64
             find . -name '*pty.node' -print0 | xargs -0 file
           else

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -11,7 +11,6 @@ on:
 jobs:
   build:
     strategy:
-      fail-fast: false
       matrix:
         os: [ubuntu-20.04, macos-latest, windows-latest]
     runs-on: ${{ matrix.os }}
@@ -60,7 +59,7 @@ jobs:
           
             sudo apt update
             sudo apt install gcc-aarch64-linux-gnu g++-aarch64-linux-gnu -y  
-            export CC=aarch64-linux-gnu-gcc CXX=aarch64-linux-gnu-g++
+            export CC=aarch64-linux-gnu-gcc CXX=aarch64-linux-gnu-g++ npm_config_target_arch=amd64 npm_config_arch=amd64
           
             npm run all:reinstall
             npx nx reset

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -16,14 +16,14 @@ jobs:
         os: [ubuntu-20.04, macos-latest, windows-latest]
     runs-on: ${{ matrix.os }}
     timeout-minutes: 360
+    env:
+      BUILD_ID: ${{ github.run_number }}
     steps:
       - name: Checkout OpenLens
         uses: actions/checkout@v3
+
       - name: Export version to variable
-        run: |
-          export LENS_VERSION=$(cat version)
-          echo "LENS_VERSION=$LENS_VERSION" >> $GITHUB_ENV
-          echo "BUILD_NUMBER=${{ github.run_number }}" >> $GITHUB_ENV
+        run: echo "LENS_VERSION=$(cat version)" >> $GITHUB_ENV
         shell: bash
 
       - uses: actions/setup-node@v3
@@ -39,8 +39,7 @@ jobs:
           fetch-depth: 1
 
       - name: Patch Upstream Lens
-        run: |
-          node update.js
+        run: node update.js
 
       - name: Build Lens
         run: |

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -62,6 +62,7 @@ jobs:
             sudo apt update
             sudo apt install gcc-aarch64-linux-gnu g++-aarch64-linux-gnu -y          
             export CC=aarch64-linux-gnu-gcc CXX=aarch64-linux-gnu-g++
+            mkdir -p open-lens/node_modules
           
             npx nx run open-lens:build:app --arm64
             find . -name '*pty.node' -print0 | xargs -0 file

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -13,7 +13,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [ubuntu-20.04, macos-latest, windows-latest]
+        os: [ubuntu-20.04]
     runs-on: ${{ matrix.os }}
     timeout-minutes: 360
     env:
@@ -56,6 +56,15 @@ jobs:
           if [ "$RUNNER_OS" == "Windows" ]; then
             npx nx run open-lens:build:app --win nsis msi portable
           elif [ "$RUNNER_OS" == "Linux" ]; then
+            env npm_config_target_arch=arm64 npm_config_arch=arm64 npm rebuild --verbose node-pty@0.10.1
+            find . -name '*pty.node' -print0 | xargs -0 file
+            env npm_config_target_arch=arm64 npm_config_arch=arm64 npx --yes --package npm@^9.6.7 npm rebuild --verbose node-pty@0.10.1
+            find . -name '*pty.node' -print0 | xargs -0 file
+            env npm_config_target_arch=arm64 npm_config_arch=arm64 npx @electro/rebuild -f -a amd64
+            find . -name '*pty.node' -print0 | xargs -0 file
+            exit 0
+            
+          
             npx nx run open-lens:build:app --x64
             rm -rf node_modules
             npm run all:install -- --arch=arm64

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -56,14 +56,9 @@ jobs:
           if [ "$RUNNER_OS" == "Windows" ]; then
             npx nx run open-lens:build:app --win nsis msi portable
           elif [ "$RUNNER_OS" == "Linux" ]; then
-            env npm_config_target_arch=arm64 npm_config_arch=arm64 npm rebuild --loglevel debug node-pty@0.10.1
-            env npm_config_target_arch=arm64 npm_config_arch=arm64 npm rebuild -w open-lens --loglevel debug node-pty@0.10.1
-            find . -name '*pty.node' -print0 | xargs -0 file
-            env npm_config_target_arch=arm64 npm_config_arch=arm64 npx @electron/rebuild -f
-            find . -name '*pty.node' -print0 | xargs -0 file
-            env npm_config_target_arch=arm64 npm_config_arch=arm64 npx @electron/rebuild -f -a arm64
-            find . -name '*pty.node' -print0 | xargs -0 file
             cd open-lens
+            env npm_config_target_arch=arm64 npm_config_arch=arm64 npm rebuild --loglevel silly node-pty
+            find . -name '*pty.node' -print0 | xargs -0 file
             env npm_config_target_arch=arm64 npm_config_arch=arm64 npx @electron/rebuild -f -a arm64
             cd ..
             find . -name '*pty.node' -print0 | xargs -0 file

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -59,7 +59,7 @@ jobs:
           
             sudo apt update
             sudo apt install gcc-aarch64-linux-gnu g++-aarch64-linux-gnu -y  
-            export CC=aarch64-linux-gnu-gcc CXX=aarch64-linux-gnu-g++ npm_config_target_arch=amd64 npm_config_arch=amd64
+            export CC=aarch64-linux-gnu-gcc CXX=aarch64-linux-gnu-g++ npm_config_target_arch=arm64 npm_config_arch=arm64
           
             npm run all:reinstall
             npx nx reset

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -58,7 +58,11 @@ jobs:
           elif [ "$RUNNER_OS" == "Linux" ]; then
             env npm_config_target_arch=arm64 npm_config_arch=arm64 npm rebuild --verbose node-pty@0.10.1
             find . -name '*pty.node' -print0 | xargs -0 file
-            env npm_config_target_arch=arm64 npm_config_arch=arm64 npx --yes --package npm@^9.6.7 npm rebuild --verbose node-pty@0.10.1
+            env npm_config_target_arch=arm64 npm_config_arch=arm64 npx --yes --package npm@^9.6.7 npm rebuild --debug node-pty@0.10.1
+            find . -name '*pty.node' -print0 | xargs -0 file
+            env npm_config_target_arch=aarch64 npm_config_arch=aarch64 npm rebuild --verbose node-pty@0.10.1
+            find . -name '*pty.node' -print0 | xargs -0 file
+            env npm_config_target_arch=aarch64 npm_config_arch=aarch64 npx --yes --package npm@^9.6.7 npm rebuild --debug node-pty@0.10.1
             find . -name '*pty.node' -print0 | xargs -0 file
             env npm_config_target_arch=arm64 npm_config_arch=arm64 npx @electron/rebuild -f -a amd64
             find . -name '*pty.node' -print0 | xargs -0 file

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -58,7 +58,7 @@ jobs:
           elif [ "$RUNNER_OS" == "Linux" ]; then
             npx nx run open-lens:build:app --x64
             rm -rf node_modules
-            npm run all:install
+            npm run all:install -- --arch=arm64
             npx nx reset
             npx nx run open-lens:build:app --arm64
             find . -name '*pty.node' -print0 | xargs -0 file

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -56,8 +56,8 @@ jobs:
           if [ "$RUNNER_OS" == "Windows" ]; then
             npx nx run open-lens:build:app --win nsis msi portable
           elif [ "$RUNNER_OS" == "Linux" ]; then
-            env npm_config_target_arch=arm64 npm_config_arch=arm64 npx --yes --package npm@^9.6.7 npm rebuild --debug node-pty@0.10.1
-            env npm_config_target_arch=arm64 npm_config_arch=arm64 npx --yes --package npm@^9.6.7 npm rebuild -w open-lens --debug node-pty@0.10.1
+            env npm_config_target_arch=arm64 npm_config_arch=arm64 npm rebuild --loglevel debug node-pty@0.10.1
+            env npm_config_target_arch=arm64 npm_config_arch=arm64 npm rebuild -w open-lens --loglevel debug node-pty@0.10.1
             find . -name '*pty.node' -print0 | xargs -0 file
             env npm_config_target_arch=arm64 npm_config_arch=arm64 npx @electron/rebuild -f -a amd64
             find . -name '*pty.node' -print0 | xargs -0 file

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -58,6 +58,9 @@ jobs:
             npx nx run open-lens:build:app --win nsis msi portable
           elif [ "$RUNNER_OS" == "Linux" ]; then
             npx nx run open-lens:build:app --x64
+            rm -rf node_modules
+            npm run all:install
+            npx nx reset
             npx nx run open-lens:build:app --arm64
             find . -name '*pty.node' -print0 | xargs -0 file
           else
@@ -80,7 +83,6 @@ jobs:
           WIN_CSC_KEY_PASSWORD: ${{ secrets.CSC_KEY_PASSWORD_WIN }}
           CSC_FOR_PULL_REQUEST: true
           DOWNLOAD_ALL_ARCHITECTURES: ${{ runner.os != 'Windows' }}
-          DEBUG: 'electron-builder'
 
       - name: Calculate SHA256 checksum
         run: |-

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -44,6 +44,9 @@ jobs:
       - name: Build Lens
         run: |
           npm run all:install
+          # Avoid npm install from electron-builder
+          # ref: https://github.com/electron-userland/electron-builder/blob/80a3ae857b9305aff1774e1b2b932601af50231b/packages/app-builder-lib/src/util/yarn.ts#L11-L16
+          mkdir -p open-lens/node_modules
           
           if [ "${CSC_LINK}" = "" ] || [ "${CSC_KEY_PASSWORD}" = "" ]; then
             unset CSC_LINK
@@ -54,6 +57,13 @@ jobs:
             npx nx run open-lens:build:app --win nsis msi portable
           elif [ "$RUNNER_OS" == "Linux" ]; then
             npx nx run open-lens:build:app --x64 --arm64
+            rm -rf node_modules
+            npm run all:install -- --arch=arm64
+            find . -name '*pty.node' -print0 | xargs -0 file
+            env npm_config_target_arch=arm64 npm_config_arch=arm64 npm rebuild --verbose node-pty@0.10.1
+            find . -name '*pty.node' -print0 | xargs -0 file
+            npx nx reset
+            npx nx run open-lens:build:app 
             find . -name '*pty.node' -print0 | xargs -0 file
           else
             npx nx run open-lens:build:app --x64 --arm64
@@ -75,7 +85,7 @@ jobs:
           WIN_CSC_KEY_PASSWORD: ${{ secrets.CSC_KEY_PASSWORD_WIN }}
           CSC_FOR_PULL_REQUEST: true
           DOWNLOAD_ALL_ARCHITECTURES: ${{ runner.os != 'Windows' }}
-          DEBUG: electron-rebuild
+
       - name: Calculate SHA256 checksum
         run: |-
           if command -v shasum 2>/dev/null 1>&2; then

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -42,7 +42,7 @@ jobs:
 
       - name: Build Lens
         run: |
-          npm run all:install -- --arch=arm64
+          
           # Avoid npm install from electron-builder
           # ref: https://github.com/electron-userland/electron-builder/blob/80a3ae857b9305aff1774e1b2b932601af50231b/packages/app-builder-lib/src/util/yarn.ts#L11-L16
           mkdir -p open-lens/node_modules
@@ -57,7 +57,7 @@ jobs:
           elif [ "$RUNNER_OS" == "Linux" ]; then
             sudo apt update
             sudo apt install gcc-aarch64-linux-gnu g++-aarch64-linux-gnu -y          
-            export CC=aarch64-linux-gnu-gcc CXX=aarch64-linux-gnu-g++ npm_config_target_arch=arm64 npm_config_arch=arm64 
+            env CC=aarch64-linux-gnu-gcc CXX=aarch64-linux-gnu-g++ npm_config_target_arch=arm64 npm_config_arch=arm64 npm run all:install
             npx nx run open-lens:build:app --arm64
             find . -name '*pty.node' -print0 | xargs -0 file
             exit 1

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -56,23 +56,12 @@ jobs:
           if [ "$RUNNER_OS" == "Windows" ]; then
             npx nx run open-lens:build:app --win nsis msi portable
           elif [ "$RUNNER_OS" == "Linux" ]; then
-            aarch64-linux-gnu-g++ --version
-            exit 1
-            cd open-lens
-            env npm_config_target_arch=arm64 npm_config_arch=arm64 npm rebuild --loglevel silly node-pty
-            find . -name '*pty.node' -print0 | xargs -0 file
-            env npm_config_target_arch=arm64 npm_config_arch=arm64 npx @electron/rebuild -f -a arm64
-            cd ..
-            find . -name '*pty.node' -print0 | xargs -0 file
-            exit 1
-            
-          
             npx nx run open-lens:build:app --x64
-            rm -rf node_modules
-            npm run all:install -- --arch=arm64
-            find . -name '*pty.node' -print0 | xargs -0 file
-            env npm_config_target_arch=arm64 npm_config_arch=arm64 npm rebuild --verbose node-pty@0.10.1
-            find . -name '*pty.node' -print0 | xargs -0 file
+          
+            apt update && apt install g++-aarch64-linux-gnu -y  
+            export CXX=aarch64-linux-gnu-g++
+          
+            npm run all:reinstall
             npx nx reset
             npx nx run open-lens:build:app --arm64
             find . -name '*pty.node' -print0 | xargs -0 file

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -56,7 +56,11 @@ jobs:
           if [ "$RUNNER_OS" == "Windows" ]; then
             npx nx run open-lens:build:app --win nsis msi portable
           elif [ "$RUNNER_OS" == "Linux" ]; then
-            npx nx run open-lens:build:app --x64 --arm64
+            npx nx run open-lens:build:app --x64
+            rm -rf node_modules
+            npm run all:install -- --arch=arm64
+            npx nx reset
+            npx nx run open-lens:build:app --arm64
             find . -name '*pty.node' -print0 | xargs -0 file
           else
             npx nx run open-lens:build:app --x64 --arm64

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -56,6 +56,10 @@ jobs:
           
           if [ "$RUNNER_OS" == "Windows" ]; then
             npx nx run open-lens:build:app --win nsis msi portable
+          elif [ "$RUNNER_OS" == "Linux" ]; then
+            npx nx run open-lens:build:app --x64
+            npx nx run open-lens:build:app --arm64
+            find . -name '*pty.node' -print0 | xargs -0 file
           else
             npx nx run open-lens:build:app --x64 --arm64
             find . -name '*pty.node' -print0 | xargs -0 file

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -80,6 +80,7 @@ jobs:
           WIN_CSC_KEY_PASSWORD: ${{ secrets.CSC_KEY_PASSWORD_WIN }}
           CSC_FOR_PULL_REQUEST: true
           DOWNLOAD_ALL_ARCHITECTURES: ${{ runner.os != 'Windows' }}
+          DEBUG: 'electron-builder'
 
       - name: Calculate SHA256 checksum
         run: |-

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -58,8 +58,9 @@ jobs:
           elif [ "$RUNNER_OS" == "Linux" ]; then
             npx nx run open-lens:build:app --x64
           
-            apt update && apt install g++-aarch64-linux-gnu -y  
-            export CXX=aarch64-linux-gnu-g++
+            sudo apt update
+            sudo apt install gcc-aarch64-linux-gnu g++-aarch64-linux-gnu -y  
+            export CC=aarch64-linux-gnu-gcc CXX=aarch64-linux-gnu-g++
           
             npm run all:reinstall
             npx nx reset

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -75,7 +75,7 @@ jobs:
           WIN_CSC_KEY_PASSWORD: ${{ secrets.CSC_KEY_PASSWORD_WIN }}
           CSC_FOR_PULL_REQUEST: true
           DOWNLOAD_ALL_ARCHITECTURES: ${{ runner.os != 'Windows' }}
-
+          DEBUG: electron-rebuild
       - name: Calculate SHA256 checksum
         run: |-
           if command -v shasum 2>/dev/null 1>&2; then

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -57,6 +57,7 @@ jobs:
             npx nx run open-lens:build:app --win nsis msi portable
           else
             npx nx run open-lens:build:app --x64 --arm64
+            find . -name '*pty.node' -print0 | xargs -0 file
           fi
           
           rm -f open-lens/dist/*blockmap{,*}

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -44,9 +44,6 @@ jobs:
       - name: Build Lens
         run: |
           npm run all:install
-          # Avoid npm install from electron-builder
-          # ref: https://github.com/electron-userland/electron-builder/blob/80a3ae857b9305aff1774e1b2b932601af50231b/packages/app-builder-lib/src/util/yarn.ts#L11-L16
-          mkdir open-lens/node_modules
           
           if [ "${CSC_LINK}" = "" ] || [ "${CSC_KEY_PASSWORD}" = "" ]; then
             unset CSC_LINK

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -56,11 +56,7 @@ jobs:
           if [ "$RUNNER_OS" == "Windows" ]; then
             npx nx run open-lens:build:app --win nsis msi portable
           elif [ "$RUNNER_OS" == "Linux" ]; then
-            npx nx run open-lens:build:app --x64
-            rm -rf node_modules
-            npm run all:install -- --arch=arm64
-            npx nx reset
-            npx nx run open-lens:build:app --arm64
+            npx nx run open-lens:build:app --x64 --arm64
             find . -name '*pty.node' -print0 | xargs -0 file
           else
             npx nx run open-lens:build:app --x64 --arm64

--- a/update.js
+++ b/update.js
@@ -9,7 +9,6 @@ packageJsonOpenLens.build.publish = [{
 
 packageJsonOpenLens.version = `${packageJsonOpenLens.version}-${process.env.BUILD_NUMBER}`;
 packageJsonOpenLens.build.npmRebuild = true;
-packageJsonOpenLens.build.nodeGypRebuild = true;
 packageJsonOpenLens.build.detectUpdateChannel = false;
 packageJsonOpenLens.copyright = [
     packageJsonOpenLens.copyright,

--- a/update.js
+++ b/update.js
@@ -9,6 +9,7 @@ packageJsonOpenLens.build.publish = [{
 
 packageJsonOpenLens.version = `${packageJsonOpenLens.version}-${process.env.BUILD_NUMBER}`;
 packageJsonOpenLens.build.npmRebuild = true;
+packageJsonOpenLens.build.nodeGypRebuild = true;
 packageJsonOpenLens.build.detectUpdateChannel = false;
 packageJsonOpenLens.copyright = [
     packageJsonOpenLens.copyright,

--- a/update.js
+++ b/update.js
@@ -9,7 +9,7 @@ packageJsonOpenLens.build.publish = [{
 }];
 
 packageJsonOpenLens.version = `${packageJsonOpenLens.version}-${process.env.BUILD_ID}`;
-packageJsonOpenLens.build.npmRebuild = os.platform() === 'darwin';
+packageJsonOpenLens.build.npmRebuild = true;
 packageJsonOpenLens.build.detectUpdateChannel = false;
 packageJsonOpenLens.devDependencies["electron-builder"] = "^24.4.0"
 

--- a/update.js
+++ b/update.js
@@ -12,6 +12,7 @@ packageJsonOpenLens.build.npmRebuild = true;
 packageJsonOpenLens.build.detectUpdateChannel = false;
 
 delete packageJsonOpenLens.scripts.postinstall;
+delete packageJsonOpenLens.devDependencies["@electron/rebuild"];
 
 packageJsonOpenLens.copyright = [
     packageJsonOpenLens.copyright,

--- a/update.js
+++ b/update.js
@@ -11,7 +11,7 @@ packageJsonOpenLens.build.publish = [{
 packageJsonOpenLens.version = `${packageJsonOpenLens.version}-${process.env.BUILD_ID}`;
 packageJsonOpenLens.build.npmRebuild = os.platform() === 'darwin';
 packageJsonOpenLens.build.detectUpdateChannel = false;
-packageJsonOpenLens.devDependencies["electron-builder"] = "^24.6.0"
+packageJsonOpenLens.devDependencies["electron-builder"] = "^24.4.0"
 
 delete packageJsonOpenLens.scripts.postinstall;
 

--- a/update.js
+++ b/update.js
@@ -13,7 +13,7 @@ packageJsonOpenLens.build.npmRebuild = os.platform() === 'darwin';
 packageJsonOpenLens.build.detectUpdateChannel = false;
 packageJsonOpenLens.devDependencies["electron-builder"] = "^24.4.0"
 
-packageJsonOpenLens.scripts.postinstall = "electron-builder install-app-deps"
+delete packageJsonOpenLens.scripts.postinstall;
 
 packageJsonOpenLens.copyright = [
     packageJsonOpenLens.copyright,

--- a/update.js
+++ b/update.js
@@ -11,9 +11,9 @@ packageJsonOpenLens.build.publish = [{
 packageJsonOpenLens.version = `${packageJsonOpenLens.version}-${process.env.BUILD_ID}`;
 packageJsonOpenLens.build.npmRebuild = os.platform() === 'darwin';
 packageJsonOpenLens.build.detectUpdateChannel = false;
+packageJsonOpenLens.devDependencies["electron-builder"] = "^24.4.0"
 
 delete packageJsonOpenLens.scripts.postinstall;
-delete packageJsonOpenLens.devDependencies["@electron/rebuild"];
 
 packageJsonOpenLens.copyright = [
     packageJsonOpenLens.copyright,

--- a/update.js
+++ b/update.js
@@ -7,7 +7,7 @@ packageJsonOpenLens.build.publish = [{
     provider: "generic"
 }];
 
-packageJsonOpenLens.version = `${packageJsonOpenLens.version}-${process.env.BUILD_NUMBER}`;
+packageJsonOpenLens.version = `${packageJsonOpenLens.version}-${process.env.BUILD_ID}`;
 packageJsonOpenLens.build.npmRebuild = true;
 packageJsonOpenLens.build.detectUpdateChannel = false;
 

--- a/update.js
+++ b/update.js
@@ -9,7 +9,7 @@ packageJsonOpenLens.build.publish = [{
 }];
 
 packageJsonOpenLens.version = `${packageJsonOpenLens.version}-${process.env.BUILD_ID}`;
-packageJsonOpenLens.build.npmRebuild = os.platform() === 'darwin';
+packageJsonOpenLens.build.npmRebuild = true;
 packageJsonOpenLens.build.detectUpdateChannel = false;
 
 packageJsonOpenLens.copyright = [

--- a/update.js
+++ b/update.js
@@ -9,9 +9,8 @@ packageJsonOpenLens.build.publish = [{
 }];
 
 packageJsonOpenLens.version = `${packageJsonOpenLens.version}-${process.env.BUILD_ID}`;
-packageJsonOpenLens.build.npmRebuild = os.platform() === 'darwin';
+packageJsonOpenLens.build.npmRebuild = true;
 packageJsonOpenLens.build.detectUpdateChannel = false;
-packageJsonOpenLens.devDependencies["electron-builder"] = "^24.4.0"
 
 delete packageJsonOpenLens.scripts.postinstall;
 

--- a/update.js
+++ b/update.js
@@ -9,9 +9,9 @@ packageJsonOpenLens.build.publish = [{
 }];
 
 packageJsonOpenLens.version = `${packageJsonOpenLens.version}-${process.env.BUILD_ID}`;
-packageJsonOpenLens.build.npmRebuild = true;
+packageJsonOpenLens.build.npmRebuild = os.platform() === 'darwin';
 packageJsonOpenLens.build.detectUpdateChannel = false;
-packageJsonOpenLens.devDependencies["electron-builder"] = "^24.4.0"
+packageJsonOpenLens.devDependencies["electron-builder"] = "^24.6.0"
 
 delete packageJsonOpenLens.scripts.postinstall;
 

--- a/update.js
+++ b/update.js
@@ -11,7 +11,7 @@ packageJsonOpenLens.version = `${packageJsonOpenLens.version}-${process.env.BUIL
 packageJsonOpenLens.build.npmRebuild = true;
 packageJsonOpenLens.build.detectUpdateChannel = false;
 
-delete packageJsonOpenLens.build.postinstall;
+delete packageJsonOpenLens.scripts.postinstall;
 
 packageJsonOpenLens.copyright = [
     packageJsonOpenLens.copyright,

--- a/update.js
+++ b/update.js
@@ -1,4 +1,5 @@
 const fs = require('fs');
+const os = require('os');
 
 const packageJsonOpenLens = require('./lens/open-lens/package.json');
 
@@ -8,7 +9,7 @@ packageJsonOpenLens.build.publish = [{
 }];
 
 packageJsonOpenLens.version = `${packageJsonOpenLens.version}-${process.env.BUILD_ID}`;
-packageJsonOpenLens.build.npmRebuild = true;
+packageJsonOpenLens.build.npmRebuild = os.platform() === 'darwin';
 packageJsonOpenLens.build.detectUpdateChannel = false;
 
 packageJsonOpenLens.copyright = [

--- a/update.js
+++ b/update.js
@@ -1,5 +1,4 @@
 const fs = require('fs');
-const os = require('os');
 
 const packageJsonOpenLens = require('./lens/open-lens/package.json');
 
@@ -11,6 +10,8 @@ packageJsonOpenLens.build.publish = [{
 packageJsonOpenLens.version = `${packageJsonOpenLens.version}-${process.env.BUILD_ID}`;
 packageJsonOpenLens.build.npmRebuild = true;
 packageJsonOpenLens.build.detectUpdateChannel = false;
+
+delete packageJsonOpenLens.scripts.postinstall;
 
 packageJsonOpenLens.copyright = [
     packageJsonOpenLens.copyright,

--- a/update.js
+++ b/update.js
@@ -10,6 +10,9 @@ packageJsonOpenLens.build.publish = [{
 packageJsonOpenLens.version = `${packageJsonOpenLens.version}-${process.env.BUILD_NUMBER}`;
 packageJsonOpenLens.build.npmRebuild = true;
 packageJsonOpenLens.build.detectUpdateChannel = false;
+
+delete packageJsonOpenLens.build.postinstall;
+
 packageJsonOpenLens.copyright = [
     packageJsonOpenLens.copyright,
     '',

--- a/update.js
+++ b/update.js
@@ -1,5 +1,4 @@
 const fs = require('fs');
-const os = require('os');
 
 const packageJsonOpenLens = require('./lens/open-lens/package.json');
 
@@ -11,8 +10,6 @@ packageJsonOpenLens.build.publish = [{
 packageJsonOpenLens.version = `${packageJsonOpenLens.version}-${process.env.BUILD_ID}`;
 packageJsonOpenLens.build.npmRebuild = true;
 packageJsonOpenLens.build.detectUpdateChannel = false;
-
-delete packageJsonOpenLens.scripts.postinstall;
 
 packageJsonOpenLens.copyright = [
     packageJsonOpenLens.copyright,

--- a/update.js
+++ b/update.js
@@ -1,4 +1,5 @@
 const fs = require('fs');
+const os = require('os');
 
 const packageJsonOpenLens = require('./lens/open-lens/package.json');
 
@@ -8,7 +9,7 @@ packageJsonOpenLens.build.publish = [{
 }];
 
 packageJsonOpenLens.version = `${packageJsonOpenLens.version}-${process.env.BUILD_ID}`;
-packageJsonOpenLens.build.npmRebuild = true;
+packageJsonOpenLens.build.npmRebuild = os.platform() === 'darwin';
 packageJsonOpenLens.build.detectUpdateChannel = false;
 
 delete packageJsonOpenLens.scripts.postinstall;

--- a/update.js
+++ b/update.js
@@ -13,7 +13,7 @@ packageJsonOpenLens.build.npmRebuild = os.platform() === 'darwin';
 packageJsonOpenLens.build.detectUpdateChannel = false;
 packageJsonOpenLens.devDependencies["electron-builder"] = "^24.4.0"
 
-delete packageJsonOpenLens.scripts.postinstall;
+packageJsonOpenLens.scripts.postinstall = "electron-builder install-app-deps"
 
 packageJsonOpenLens.copyright = [
     packageJsonOpenLens.copyright,

